ui/src/App.jsx

```diff
@@ -42,10 +43,12 @@ class App extends Component {
 
   sendUpdate() {
     const state = JSON.parse(JSON.stringify(this.props.state));
-    for (let sp of state.Subpolicies) {
-      sp.Metadata = JSON.parse(sp.Metadata);
-    }
-    axios.post(`${window.config.env.REACT_APP_NOPAS_ENDPOINT}/update`, state);
+    const out = uiToServerConversion(state);
```

Comment:
I would suggest putting the cloning of `state` within the `uiToServerConversion` function, so that it wouldn't impose a restriction on the user of `uiToServerConversion` to create a clone beforehand.

Reply:
yeap good idea